### ISSUE TITLE
Unable To Set ssl.endpoint.identification.algorithm= to null

### ISSUE THREAD

**Issue by eddideku** [bitnami/bitnami-docker-kafka#39]
Hello,

For self signed certificates and local development I am running into SSL Handshake issues when attempting to auth a producer/consumer. Is there kafka security documentation on how to get official signed CA certs into the dev environment?

`
[2019-01-24 08:51:52,412] ERROR [Controller id=1001, targetBrokerId=1001] Connection to node 1001 (3590c068b260/1.0.0.4:9092) failed authentication due to: SSL handshake failed (org.apache.kafka.clients.NetworkClient)                                                    
[2019-01-24 08:51:52,564] ERROR [Controller id=1001, targetBrokerId=1001] Connection to node 1001 (3590c068b260/1.0.0.4:9092) failed authentication due to: SSL handshake failed (org.apache.kafka.clients.NetworkClient)
`

I don't see anything in README.md that allows to switch ssl.endpoint.identification.algorithm= to null

Hostname verification is by default turned on (https) and I am unaware of how to properly set this up with self signed certs. (I'm new to Kafka). I have to shell into the containers and manually insert this line. Not ideal when constantly adjusting the docker-compose file.

```
echo "ssl.endpoint.identification.algorithm=" >> /opt/bitnami/kafka/conf/server.properties 
echo "ssl.endpoint.identification.algorithm=" >> /opt/bitnami/kafka/conf/producer.properties 
echo "ssl.endpoint.identification.algorithm=" >> /opt/bitnami/kafka/conf/consumer.properties 
```

**Comment by marcosbc** [comment]
Hi @eddideku, could you describe what steps you performed in order to enable SSL?

Note that we have some instructions on how to do this, particularly check the "Security" section in the README: https://github.com/bitnami/bitnami-docker-kafka#Security

Note that you must set up the Java Key Store (JKS), and we mention a script that automatically does this.
After that, you must locate the key store in `/opt/bitnami/kafka/conf/certs`. Also, if your JKS has a password, you must configure Kafka take it into account.

**Comment by eddideku** [comment]
> Hi @eddideku, could you describe what steps you performed in order to enable SSL?
> 
> Note that we have some instructions on how to do this, particularly check the "Security" section in the README: https://github.com/bitnami/bitnami-docker-kafka#Security
> 
> Note that you must set up the Java Key Store (JKS), and we mention a script that automatically does this.
> After that, you must locate the key store in `/opt/bitnami/kafka/conf/certs`. Also, if your JKS has a password, you must configure Kafka take it into account.

Hello @marcosbc I believe I solved my own issue, which it wasn't apparent in the documentation. I kept getting SSL Handshake issues following the documentation. My lack of knowledge with java keystore, keytool certs and docker-compose may have limited myself.

I recreated a single node kafka docker-compose.yml based on the examples in the documentation. However, I did not realize while generating the certificates, the Common Name (in the first step generate-ssh.sh uses openssl) should be my FQDN (ie in my environment kafka.example.com) and a few steps in the bash script it uses keytool where it prompts "First and Lastname" (ie I used kafka.example.com).

Now the next big thing was my lack of knowledge with docker. A colleague suggested in docker-compose.yml to specify a hostname. Example,

`
kafka:
  hostname: kafka.example.com
  image: 'bitnami/kafka:latest'
  ports:
    - '9092'
  environment:
... (you get the picture)
`

Once I did this, I mounted the keystore and truststore as you suggested

`
volumes:
  - './keystore/kafka.keystore.jks:/opt/bitnami/kafka/conf/certs/kafka.keystore.j ks:ro'                                                                                              
  - './truststore/kafka.truststore.jks:/opt/bitnami/kafka/conf/certs/kafka.truststore.jks:ro'
`

The only problem I'm running into right now is that consumer.properties and producer.properties have the `bootstrap.servers=localhost:9092`

I manually went in and changed that to `bootstrap.server=kafka.example.com:9092`

sed -i 's/localhost:9092/kafka.example.com:9092/g`

Obviously that's not the most elegant solution.

One thing to note, I also set my DNS for kafka.example.com to my ubuntu server. I cannot get a local VM to connect to my ec2 instance.
